.circleci/config.yml

```diff
@@ -211,6 +259,26 @@ workflows:
                 - "./gemfiles/sinatra_14.gemfile"
                 - "./gemfiles/shoryuken_50.gemfile"
                 - "./gemfiles/mongo_216.gemfile"
+            exclude:
+            # Currently rest-client 1.6 gemfile pulls in mime-types 2.99.3
+            # and adaptation to mime-type 3.0 is only available in rest-client 2.0:
+            # https://github.com/rest-client/rest-client/issues/464
+            # But mime-types 2.99.3 is too old for Ruby 3.0 or newer:
+            # https://stackoverflow.com/a/68433864
+            # so we should exclude thiese scenarios
+            - stack: ruby_30
+              gemfile: "./gemfiles/rest_client_16.gemfile"
+            - stack: ruby_31
+              gemfile: "./gemfiles/rest_client_16.gemfile"
+            - stack: ruby_32
+              gemfile: "./gemfiles/rest_client_16.gemfile"
+            # Sinatra 1.4 relies on the Fixnum class
+            # which has been removed from Ruby 3.2, see:
+            # https://bugs.ruby-lang.org/issues/12005
+            # https://www.ruby-lang.org/en/news/2022/11/11/ruby-3-2-0-preview3-released/
+            - stack: ruby_32
+              gemfile: "./gemfiles/sinatra_14.gemfile"
+
```

Comment:
With the adding support to Ruby 3.0, 3.1, and 3.2, wouldn't it be interesting to support the latest (or, at least, higher) version of the frameworks and clients to prevent them from having these dependencies problems?

Reply:
Supporting the latest version of the framework doesn't have an impact on version incompatibilities in the current tests matrix.
It's the dropping of the older versions from the test matrix, which would have.
That is something that we should consider after a while when the exclude list is growing out of control,
or when the framework version is known to be EOL or vulnerable etc.
We have dropped older versions in the past anyway.
My main point is, that even if we drop official support for an older framework version,
we still at least have to verify that instana tracing does not crash things, that is the goal of the robustness initiative.

BTW, from `rest-client` we already support the latest, and sinatra is definitely on the radar, only not the topic of this PR.